tools/scan-build-py/libscanbuild/analyze.py

```diff
@@ -544,13 +549,16 @@ def generate_ast(triple_arch):
         """ Generates ASTs for the current compilation command. """
 
         args = opts['direct_args'] + opts['flags']
-        ast_joined_path = os.path.join(opts['ctu'].dir, 'ast', triple_arch,
+        ast_joined_path = os.path.join(opts['ctu'].dir, triple_arch, 'ast',
                                        os.path.realpath(opts['file'])[1:] +
                                        '.ast')
         ast_path = os.path.abspath(ast_joined_path)
         ast_dir = os.path.dirname(ast_path)
         if not os.path.isdir(ast_dir):
-            os.makedirs(ast_dir)
+            try:
+                os.makedirs(ast_dir)
+            except OSError:
```

Comment:
Is it a good idea to pass on every exception? I guess you only want to pass if the dir already exists and fail otherwise.

Reply:
This was the solution in order to be compatible with python 3. You can get an exception on having different mode of the dir compared to the default 777 of this command. I don't know about any other useful exception to keep here.